src/redux/Books/bookSlice.jsx

```diff
@@ -0,0 +1,43 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  bookItems: [
+    {
+      item_id: 'item1',
+      title: 'The Great Gatsby',
+      author: 'John Smith',
+      category: 'Fiction',
+    },
+    {
+      item_id: 'item2',
+      title: 'Anna Karenina',
+      author: 'Leo Tolstoy',
+      category: 'Fiction',
+    },
+    {
+      item_id: 'item3',
+      title: 'The Selfish Gene',
+      author: 'Richard Dawkins',
+      category: 'Nonfiction',
+    },
+  ],
+};
+
+const bookSlice = createSlice({
+  name: 'books',
+  initialState,
+
+  reducers: {
+    addBook: (state, action) => ({
+      ...state,
+      book: [...state, action.payload],
+    }),
+    removeBook: (state, action) => ({
+      ...state,
+      book: state.books.filter((book) => book.id !== action.payload),
+    }),
```

Comment:
- [ ] In both of your reducers, you are returning a new state of `book` with the payload attached or removed depending upon the reducer called. However, in the initial state, the `book` array doesn't exist. Instead, it is a `bookItems` array. I would recommend renaming either one to the other. Preferably naming your array as `books`.
- [ ] The id property you are trying to access on the `book` in the `removeBook` reducer doesn't exist in the initialState. Instead, it is `item_id`. Kindly change either one to the other.